lib/ruby_lsp/standard/wraps_built_in_lsp_standardizer.rb

```diff
@@ -0,0 +1,83 @@
+module RubyLsp
+  module Standard
+    class WrapsBuiltinLspStandardizer
+      include RubyLsp::Requests::Support::Formatter
+      def initialize
+        init!
+      end
+
+      def init!
+        @config = ::Standard::BuildsConfig.new.call(ARGV)
+        @standardizer = ::Standard::Lsp::Standardizer.new(
+          @config,
+          ::Standard::Lsp::Logger.new
+        )
+        @cop_registry = RuboCop::Cop::Registry.global.to_h
+      end
+
+      def run_formatting(uri, document)
+        @standardizer.format(uri_to_path(uri), document.source)
```

Comment:
We have a helper that patches a new method into URI objects to get the correct file system paths, which has some fixes for Windows (the URI gem has some bugs).

```suggestion
        @standardizer.format(uri.to_standardized_path, document.source)
```

There's only one caveat. If you created a Ruby file and it is currently unsaved, you can invoke formatting manually. In those cases, `to_standardized_path` returns `nil`. I'm not sure how Standard will handle that.

Reply:
I suppose I'll just wrap my own helper and guard against the nil case to fall back:

```ruby
      def uri_to_path(uri)
        if uri.respond_to?(:to_standardized_path) && !(standardized_path = uri.to_standardized_path).nil?
          standardized_path
        else
          uri.to_s.sub(%r{^file://}, "")
        end
      end
```